txtorcon/circuit.py

```diff
@@ -275,3 +275,25 @@ def __str__(self):
         path = ' '.join([x.ip for x in self.path])
         return "<Circuit %d %s [%s] for %s>" % (self.id, self.state, path,
                                                 self.purpose)
+
+class CircuitBuildTimedOutError(Exception):
+    """
+    This exception is thrown when using `timed_circuit_build`
+    and the circuit build times-out.
+    """
+
+def build_timeout_circuit(tor_state, clock, path, timeout):
+    """
```

Comment:
What about just calling this ``build_circuit`` and make timeout optional? i.e. ``build_circuit(tor_state, clock, path, timeout=None)``.

Kind of "bikeshed" territory, but I also think it would be more consistent to call the ``clock`` arg ``reactor`` instead?